### DISCUSSION TITLE
Is the test bootstrapper app still needed?

### DISCUSSION THREAD

**Discussion by Numpsy** [icsharpcode/SharpZipLib#630] [Ideas] [answered]
Just a thought - is the test bootstrapper app still needed, or can it be removed now that everything is done via Github actions?

**Comment by piksel** [comment, marked as answer]
The only reason for keeping it right now was in case it would be needed to get better coverage output, but I actually think I just found what causes the code coverage diff....
![image](https://user-images.githubusercontent.com/807383/117976831-7d7bce80-b330-11eb-9347-9d0c806cf8cf.png)

The tests are using random data:
https://github.com/icsharpcode/SharpZipLib/blob/232d690e4a01a1483f07c78572ecae58849900c1/test/ICSharpCode.SharpZipLib.Tests/BZip2/Bzip2Tests.cs#L25-L27

...which means that it's random whether all code paths are followed... 😅

https://en.wikipedia.org/wiki/Bzip2#Run-length_encoding_of_MTF_result

**Reply by piksel** [reply]
Should be fixed in #636 and I see no reason to keep the bootstrapper anymore.
